### PR TITLE
[FEAT] add thread-safe code for implicit loading of spyre-inductor

### PR DESCRIPTION
**Pull request by thoangtrvn**
Thanks for sending a pull request! Please make sure you read the [contributing guidelines](https://github.com/torch-spyre/torch-spyre/blob/main/CONTRIBUTING.md).

#### What type of PR is this?

- [x] bug
- [x] feature
- [ ] documentation
- [ ] cleanup

#### What this PR does:

Context / Related: Decomposition work in PR #535 changed when and where Spyre Inductor initialization happens. During review (see discussion on that PR), we observed that removing preload.py and relocating its logic shifted initialization timing, which surfaced new ordering issues. E.g. (https://github.com/torch-spyre/torch-spyre/pull/535#discussion_r2790905836)

Spyre’s runtime requires exclusive device ownership. Initializing too early (e.g., at import torch) can block unrelated PyTorch sessions; initializing too late causes compile‑time failures. Current explicit loading of spyre inductor currently relies on running 
```
torch.manual_seed(0xAFFE)
```
or creating a spyre tensor prior running spyre inductor. 

GOAL: Keep lazy loading (avoid claiming hardware unless Spyre is actually used), but ensure Spyre’s Inductor plumbing is in place exactly when needed—without forcing users to call explicit setup code or seed as a workaround.

Here, we introduce a thread-safe mechanism that reliably detect the target compile device, e.g. 'spyre', and maintain the lazy-loading mechanism which will ensure the code to work when changes are introduced in PR #535. 

```python
import torch
def test(end, device=None):
        return torch.arange(end, device=device, dtype=torch.float16)

compiled = torch.compile(test, backend="inductor")
output = compiled(64.0,  device="spyre")

output_cpu = output.cpu()

```

#### Which issue(s) this PR is related to:

PR #535. 

Issue #641 

Issue #472 

#### Additional note:

CC: @bohnstingl, @yoheiueda.

### CODE REVIEW

**Comment by thoangtrvn** [comment]
bot:test

**Review by bohnstingl — CHANGES_REQUESTED** [review]
Thank you @thoangtrvn for the PR. I tested it and observed some minor points that we might want to address.

**Review by yoheiueda — DISMISSED** [review]
LGTM. Thank you very much!

**Review by dgrove-oss — DISMISSED** [review]
lgtm

File: torch_spyre/_inductor/__init__.py [hunk 1]
```diff
@@ -14,98 +14,155 @@
...
+            # Graph nodes (covers tensorless factories)
+            for n in gm.graph.nodes:
+                dev = n.kwargs.get("device")
+                if dev is None:
+                    continue
+                if isinstance(dev, torch.device) and dev.type == device_name:
+                    return True
+                if isinstance(dev, str) and dev.split(":")[0] == device_name:
+                    return True
+            return False
```
[the remark this answers is not in the document]
Comment (thoangtrvn) [reply torch_spyre/_inductor/__init__.py]: @bohnstingl : Can you check: the fast-track code path is added now.

**Review by bohnstingl — APPROVED** [review]

**Comment by thoangtrvn** [comment]
bot:test
